src/qutip_jax/linalg.py

```diff
@@ -13,21 +13,24 @@
 ]
 
 
-@partial(jit, static_argnums=[1, 2, 3, 4])
+@partial(jit, static_argnums=[2, 3, 4])
 def _eigs_jaxarray(data, isherm, vecs, eigvals, low_first):
     """
     Internal function to dispatch the eigenvalue solver to `eigh`, `eig`,
     `eigvalsh` or `eigvals` based on the parameters.
     """
-    if isherm and vecs:
-        evals, evecs = jnp.linalg.eigh(data)
-    elif vecs:
-        evals, evecs = jnp.linalg.eig(data)
-    elif isherm:
-        evals = jnp.linalg.eigvalsh(data)
-        evecs = None
+    if vecs:
+        evals_herm, evecs_herm = jnp.linalg.eigh(data)
+        evals_nonherm, evecs_nonherm = jnp.linalg.eig(data)
```

Comment:
The computation of eigen values is slow. We use the hermitian version in part because it's less slow than the non-hermitian one.
Doing it the way will kill efficiency...
I think we can do better.

Reply:
I tried different techniques like `jax.lax.cond` but that requires the output to be of same dimensions that doesn't match with this case.